nix/shell-plugins.nix

```diff
@@ -0,0 +1,57 @@
+{ pkgs, lib, config, is-home-manager, ... }:
+with lib;
+let cfg = config.programs._1password-shell-plugins;
+in {
+  options = {
+    programs._1password-shell-plugins = {
+      enable = mkEnableOption "1Password Shell Plugins";
+      plugins = mkOption {
+        type = types.listOf types.package;
+        default = [ ];
+        example = literalExpression ''
+          with pkgs; [
+            gh
+            awscli2
+            cachix
+          ]
+        '';
+        description =
+          "CLI Packages to enable 1Password Shell Plugins for; ensure that a Shell Plugin exists by checking the docs: https://developer.1password.com/docs/cli/shell-plugins/";
+      };
+    };
+  };
+
+  config = let
+    # executable names as strings, e.g. `pkgs.gh` => `"gh"`, `pkgs.awscli2` => `"aws"`
+    pkg-exe-names = map (package:
+      # NOTE: SAFETY: This is okay because the `packages` list is also referred
+      # to below as `home.packages = packages;` or `environment.systemPackages = packages;`
+      # depending on if it's using `home-manager` or not; this means that Nix can still
+      # compute the dependency tree, even though we're discarding string context here,
+      # since the packages are still referred to below without discarding string context.
+      builtins.unsafeDiscardStringContext (baseNameOf (getExe package)))
+      cfg.plugins;
+    # Explanation:
+    # Map over `cfg.plugins` (the value of the `plugins` option provided by the user)
+    # and for each package specified, get the executable name, then create a shell alias
+    # of the form:
+    # `alias {pkg}="op plugin run -- {pkg}"`
+    # where `{pkg}` is the executable name of the package
+    aliases = listToAttrs (map (package: {
+      name = package;
+      value = "op plugin run -- ${package}";
+    }) pkg-exe-names);
```

Comment:
This results in an `attrset` in the form:

```nix
aliases = {
  gh = "op plugin run -- gh";
  aws = "op plugin run -- aws";
  cachix = "op plugin run -- cachix";
};
```

Using a Nix `attrset` for the option (see `bash.shellAliases = aliases;` below) allows us to not depend on the shell's specific alias definition syntax, so if we ever decided to support additional shells like `nushell` or something, it would only require an additional attribute below, we wouldn't have to modify the way the aliases are created (since Nix handles actually setting them in the shell config for us).